### PR TITLE
[libc++] Remove linux Buildkite builders entirely

### PR DESCRIPTION
**Pull request by EricWF**
This removes the Google hosted Linux buildkite builders. We have since moved all of them over to github actions.

Follow up changes will be sent for android.

### CODE REVIEW

**Review by EricWF — COMMENTED** [review]
I plan to commit this once the bots come back green.

File: libcxx/utils/ci/buildkite-pipeline.yml [hunk 1]
```diff
@@ -101,20 +101,6 @@ environment_definitions:
 
 
 steps:
-- label: Bootstrapping build
```
Comment (ldionne) [inline libcxx/utils/ci/buildkite-pipeline.yml:104]: I think we can remove the `linux_agent` definition above now!

**Review by ldionne — COMMENTED** [review]
Thanks a lot for the cleanup, this was on my list! This should allow dropping the GCE runners on Linux altogether, right? Might be worth recording that benefit in the commit message.

File: libcxx/utils/ci/buildkite-pipeline.yml (hunk 1 above)
Comment (EricWF) [reply libcxx/utils/ci/buildkite-pipeline.yml:104]: Great!

Setting up the Mac OS X bots should be really easy too! Let me know if you want some help.

**Comment by EricWF** [comment]
Submitting before the bots come back so I can stop wasting resources.

**Review by ldionne — APPROVED** [review]
LGTM, thanks!

**Pull request merged**